server/etcdmain/config.go

```diff
@@ -238,6 +239,30 @@ func (cfg *config) configFromCmdLine() error {
 		cfg.ec.InitialCluster = ""
 	}
 
+	// getBoolFlagVal checks if the a given flag is explicitly set in the cmd line arguments.
+	getBoolFlagVal := func(flagName string) (bool, bool) {
```

Comment:
```suggestion
	getBoolFlagVal := func(flagName string) *bool {
```

Three value logic (true, false, unset) here would better map to pointer instead of two unnamed bool